xarray/core/dataset.py

```diff
@@ -306,7 +306,7 @@ class Dataset(Mapping, ImplementsDatasetReduce, BaseDataObject,
     groupby_cls = groupby.DatasetGroupBy
 
     def __init__(self, data_vars=None, coords=None, attrs=None,
-                 compat='broadcast_equals'):
+                 compat='broadcast_equals', encoding=None):
```

Comment:
I think consistency makes for a good argument here, though I would consider deprecating the `encoding` argument to `DataArray` instead. It would also make sense to get rid of the `compat` argument to `Dataset`.

These extra arguments are not part of the fundamental xarray data model and thus are a little distracting, especially to new users.